shenyu-admin/src/main/java/org/apache/shenyu/admin/config/LdapConfiguration.java

```diff
@@ -47,17 +48,19 @@
     @Bean
     @ConditionalOnMissingBean
     public LdapContextSource contextSource(final LdapProperties ldapProp) {
-        LdapContextSource contextSource = new LdapContextSource();
-        contextSource.setUrl(ldapProp.getUrl());
-        contextSource.setUserDn(ldapProp.getBindDn());
-        contextSource.setPassword(ldapProp.getPassword());
-        contextSource.setPooled(true);
-        Map<String, Object> config = new HashMap<>();
-        config.put("java.naming.ldap.attributes.binary", "objectGUID");
-        config.put("com.sun.jndi.ldap.connect.timeout", String.valueOf(ldapProp.getConnectTimeout()));
-        config.put("com.sun.jndi.ldap.read.timeout", String.valueOf(ldapProp.getReadTimeout()));
-        contextSource.setBaseEnvironmentProperties(config);
-        return contextSource;
+        return Optional.ofNullable(ldapProp).map(item -> {
```

Comment:
Parameter `ldapprop` may be null?

Reply:
I think it's possible.

Reply:
why?It has been registerd by `@EnableConfigurationProperties(LdapProperties.class)`

Reply:
I don't understand the function of this annotation. I've learned it now. Thank you.